### PR TITLE
feat: update parameter path from `sdk` to `manta-parameters`

### LINKED ISSUE #605: Move to new Public Parameter Location
We are moving public parameters from `sdk` to `manta-rs`. This requires a dependency change in `pallet-manta-pay`.

### PR DESCRIPTION
**Pull request by BoyuanFeng**
Signed-off-by: Boyuan Feng <bfeng9@wisc.edu>

## Description

In `manta-rs v0.5.1`, we have moved public parameters from `sdk` to `manta-parameters`. This requires a dependency change in `pallet-manta-pay` and a tag update for `manta-rs` crates from `v0.5.0` to `v0.5.1`.

We also update the random number generator from `thread_rng` to `OsRng`, removing an unnecessary `rand` dependency. In one other case, where we need deterministic RNG, we use fixed-seed ChaCha-based RNG and import the `rand_chacha` crate for this. (The `rand_chacha` crate is the canonical ChaCha implementation in Rust and is used in `manta-rs`).

closes: #605 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [x] Wrote unit tests.
- [x] Updated relevant documentation in the code.
- [x] Added **one** line describing your change in `<branch>/CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer.
- [x] If runtime changes, need to update the version numbers properly:
   * `authoring_version`: The version of the authorship interface. An authoring node will not attempt to author blocks unless this is equal to its native runtime.
   * `spec_version`: The version of the runtime specification. A full node will not attempt to use its native runtime in substitute for the on-chain Wasm runtime unless all of spec_name, spec_version, and authoring_version are the same between Wasm and native.
   * `impl_version`: The version of the implementation of the specification. Nodes are free to ignore this; it serves only as an indication that the code is different; as long as the other two versions are the same then while the actual code may be different, it is nonetheless required to do the same thing. Non-consensus-breaking optimizations are about the only changes that could be made which would result in only the impl_version changing.
   * `transaction_version`: The version of the extrinsics interface. This number must be updated in the following circumstances: extrinsic parameters (number, order, or types) have been changed; extrinsics or pallets have been removed; or the pallet order in the construct_runtime! macro or extrinsic order in a pallet has been changed. You can run the `metadata_diff.yml` workflow for help. If this number is updated, then the `spec_version` must also be updated 
- [x] Verify benchmarks & weights have been updated for any modified runtime logics
- [x] If importing a new pallet, choose a proper module index for it, and allow it in `BaseFilter`. Ensure **every** extrinsic works from front-end. If there's corresponding tool, ensure both work for each other.
- [x] If needed, update our Javascript/Typescript APIs. These APIs are officially used by exchanges or community developers.
- [x] If modifying existing runtime storage items, make sure to implement storage migrations for the runtime and test them with `try-runtime`. This includes migrations inherited from upstream changes, and you can search the diffs for modifications of `#[pallet::storage]` items to check for any.

### CODE REVIEW

**Review by bhgomes — DISMISSED** [review]

**Review by bhgomes — APPROVED** [review]

File: Cargo.lock
Comment (Garandor) [inline Cargo.lock:6653]: 0.3 is a fairly low version number - the crate is only a year old. Are you sure this is safe to depend on?

File: pallets/manta-pay/Cargo.toml
Comment (Garandor) [inline pallets/manta-pay/Cargo.toml:102]: not sure if you already do so, but the version bump here is not described in the PR or changelog. Maybe it should be

File: pallets/manta-pay/Cargo.toml
Comment (Garandor) [inline pallets/manta-pay/Cargo.toml:71]: also, changing the RNG is not described in changelog or PR

**Review by Garandor — COMMENTED** [review]
minor description nits and a question about safety of switching to a new-ish crate

File: Cargo.lock
Comment (BoyuanFeng) [reply Cargo.lock:6653]: `rand_chacha` is the best implementation for chacha. We have already used it in `manta-rs`. I will add a description in PR.

File: pallets/manta-pay/Cargo.toml
Comment (BoyuanFeng) [reply pallets/manta-pay/Cargo.toml:71]: Will add in PR.

File: pallets/manta-pay/Cargo.toml
Comment (BoyuanFeng) [reply pallets/manta-pay/Cargo.toml:102]: Thanks for pointing out! I have updated in the PR description.

File: Cargo.lock
Comment (Garandor) [reply Cargo.lock:6653]: > We're already using it elsewhere 
> is the best implementation

are not really satisfactory answers to the "is it safe?" question ;)
I was hoping to hear you guys looked at the code yourselves or know of at least 5 other substrate projects that use it too.

I guess I'll defer to @bhgomes who already approved and go along if Brandon confirms y'all are aware of this risk.

Comment (bhgomes) [reply Cargo.lock:6653]: Yeah the ChaCha code is correct, it's very easy to audit, it's nothing like auditing `rand` itself, good luck actually auditing that beast. As for trust surface, it's just used as an RNG for generating dummy data used to make sure that changes to manta-pay don't break binary compatibility with the current ledger state. It's not used by the runtime logic anywhere.

Also the crate is actually 3 years old and [maintained by the `rand` group itself](https://github.com/rust-random/rand/tree/master/rand_chacha).

**Review by Garandor — APPROVED** [review]

**Pull request merged**
